### PR TITLE
build helm charts and reorganize demo dir

### PR DESCRIPTION
**Pull request by bassam**

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -318,6 +318,11 @@ imports:
   subpackages:
   - pkg/apis/apiextensions
   - pkg/apis/apiextensions/v1beta1
+  - pkg/client/clientset/clientset
```
Comment (jbw976) [inline glide.lock]: the glide.yaml didn't add new packages, do you know why these new dependencies got added to the glide.lock?

File: build/makelib/helm.mk [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+define helm.chart
+$(HELM_OUTPUT_DIR)/$(1)-$(VERSION).tgz: $(HELM) $(HELM_OUTPUT_DIR) $(shell find $(HELM_CHARTS_DIR)/$(1) -type f)
+	@echo === helm package $(1)
+	@cp -f $(HELM_CHARTS_DIR)/$(1)/values.yaml.tmpl $(HELM_CHARTS_DIR)/$(1)/values.yaml
+	@cd $(HELM_CHARTS_DIR)/$(1) && $(SED_CMD) 's|%%VERSION%%|$(VERSION)|g' values.yaml
+	@$(HELM) lint --strict $(abspath $(HELM_CHARTS_DIR)/$(1))
+	@$(HELM) package --version $(VERSION) -d $(HELM_OUTPUT_DIR) $(abspath $(HELM_CHARTS_DIR)/$(1))
+$(HELM_INDEX): $(HELM_OUTPUT_DIR)/$(1)-$(VERSION).tgz
+endef
+$(foreach p,$(HELM_CHARTS),$(eval $(call helm.chart,$(p))))
```
Comment (jbw976) [inline build/makelib/helm.mk]: what other helm charts do we expect to support in the future?

File: glide.lock (hunk 1 above)
Comment (bassam) [inline glide.lock]: not sure. they got picked up after I did a `make vendor`

File: build/makelib/helm.mk (hunk 2 above)
Comment (bassam) [inline build/makelib/helm.mk]: possible #710 and maybe others. I'm not sure yet.

**Comment by bassam** [comment]
I'll deal with versioning the demo yaml files separately.

**Pull request merged**